manim/mobject/types/vectorized_mobject.py

```diff
@@ -1402,13 +1706,28 @@ def get_anchors(self) -> np.ndarray:
         """
         if self.points.shape[0] == 1:
             return self.points
-        return np.array(
-            list(it.chain(*zip(self.get_start_anchors(), self.get_end_anchors()))),
-        )
+        num_curves = self.get_num_curves()
+        anchors = np.empty((2 * num_curves, self.dim))
+        anchors[0::2] = self.get_start_anchors()[:num_curves]
+        anchors[1::2] = self.get_end_anchors()
```

Comment:
Why is the slice required for the start anchors but not the end anchors?

Reply:
Because I've seen in the source code that there are some cases where `self.points` ends with a single start anchor (or straight up has a single point in general) instead of a full quad representing a Bézier curve. In these cases, these lone points do not increase the `num_curves` and can be stripped off.

However, when there's an end anchor ending a Bézier curve, this also means there's another Bézier curve to consider in `num_curves`.

What I mean is, there are as many Bézier curves as there are end anchors, and thus there's no need to slice these end anchors. However, not every start anchor means the presence of another Bézier curve, and thus the start anchors **must** be sliced, just in case.

But yeah, this might be confusing for other people looking at the source code, so I should add an explanationy comment in the code (or straight up slice both arrays of anchors, even if the end anchors do not really need it).

Reply:
Added the corresponding comment.